test/functionalTest/cases/1112_metadata_as_keymap/metadata_with_dots_v2.test

```diff
@@ -0,0 +1,122 @@
+#
+# This file is part of Orion Context Broker.
+#
+# Orion Context Broker is free software: you can redistribute it and/or
+# modify it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# Orion Context Broker is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero
+# General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with Orion Context Broker. If not, see http://www.gnu.org/licenses/.
+#
+# For those usages not covered by this license please contact with
+# iot_support at tid dot es
+
+# VALGRIND_READY - to mark the test ready for valgrindTestSuite.sh
+
+--NAME--
+Metadata with dot in the name (NGSIv2)
+
+--SHELL-INIT--
+dbInit CB
+brokerStart CB
+
+--SHELL--
+
+# 01. Create entity with metadata with dot in the name
+# 02. Retieve entity
+# 03. Check the '=' in DB
+
+echo "01. Create entity with metadata with dot in the name"
+echo "===================================================="
+url="/v2/entities"
+payload='{
+  "id": "E",
+  "type": "T",
+  "A": {
+    "value": "foo",
+    "type": "T",
+    "metadata": {
+      "m.x": {
+        "type": "string",
+        "value": "black"
+      },
+      "m.y": {
+        "type": "double",
+        "value": 5
+      }
+    }
+  }
+}'
+orionCurl --url "$url" --payload "${payload}"
+echo
+echo
+
+
+echo "02. Retieve entity"
+echo "=================="
+orionCurl --url "/v2/entities/E"
+echo
+echo
+
+
+echo "03. Check the '=' in DB"
+echo "======================="
+mongoCmd ${CB_DB_NAME} 'db.entities.find({ "_id" : { "id" : "E",  "type": "T", "servicePath" : "/" } }, {_id:0, attrs: 1})'
+echo
+echo
+
+
+--REGEXPECT--
+01. Create entity with metadata with dot in the name
+====================================================
+HTTP/1.1 201 Created
+Content-Length: 0
+Location: /v2/entities/E?type=T
+Fiware-Correlator: REGEX([0-9a-f\-]{36})
+Date: REGEX(.*)
+
+
+
+02. Retieve entity
+==================
+HTTP/1.1 200 OK
+Content-Length: 139
+Content-Type: application/json
+Fiware-Correlator: REGEX([0-9a-f\-]{36})
+Date: REGEX(.*)
+
+{
+    "A": {
+        "metadata": {
+            "m.x": {
+                "type": "string",
+                "value": "black"
+            },
+            "m.y": {
+                "type": "double",
+                "value": 5
+            }
+        },
+        "type": "T",
+        "value": "foo"
+    },
+    "id": "E",
+    "type": "T"
+}
+
+
+03. Check the '=' in DB
+=======================
+{ "attrs" : { "A" : { "type" : "T", "creDate" : REGEX(.*), "value" : "foo", "md" : { "m=x" : { "type" : "string", "value" : "black" }, "m=y" : { "type" : "double", "value" : 5 } }, "mdNames" : [ "m.x", "m.y" ] } } }
```

Comment:
What is important to review here is that the key in the `md` maps correctly uses `=` as encoding character for dots.